pkg/workloadmeta/collectors/internal/remote/process_collector/process_collector.go

```diff
@@ -123,10 +130,6 @@ func (s *streamHandler) Port() int {
 	return s.port
 }
 
-func (s *streamHandler) IsEnabled() bool {
-	return config.Datadog.GetBool("workloadmeta.remote_process_collector.enabled")
```

Comment:
Why can't we do:
```
	if flavor.GetFlavor() != flavor.DefaultAgent {
		return
	}
```
in the `IsEnabled` function?

I think this might be simpler to comprehend tbh.

Reply:
I prefer to follow the same pattern as other collectors. This way, we have only one way to select components. 

However, I just realized that I forgot `workloadmeta.remote_process_collector.enabled` in Start function

Reply:
Actually, nice catch! I realized that `GetFlavor` shouldn't be called in init functions.